File: Java_Solutions/Find_the_Index_of_the_First_Occurrence_in_a_String.java
/*
* Leetcode Challenge: Find the Index of the First Occurence in a String
*
* Given two strings needle and haystack, return the index of the first occurrence of 
* needle in haystack, or -1 if needle is not part of haystack.
* 
* 5/29/24
* Mark Robuck
*/
class Solution {
    /*
    * find the index of the first time needle appears in haystack
    * @params haystack, string to be searched
    * @params needle, string being searched for
    * @return, index of the first occurrence of needle in haystack as an int
    */
    public int strStr(String haystack, String needle) {
        return haystack.indexOf(needle);
    }
}

File: Python_Solutions/Find_the_Index_of_the_First_Occurrence_in_a_String.py
"""
Leetcode Challenge: Find the Index of the First Occurrence in a String

Given two strings needle and haystack, return the index of the first occurrence of 
needle in haystack, or -1 if needle is not part of haystack.
 
5/29/24
Mark Robuck
"""
class Solution:
    """
    find the index of the first time needle appears in haystack
    Args: 
        haystack(string): string to be searched
        needle(string): string being searched for
    Returns:
        int: index of the first occurrence of needle in haystack
    """
    def strStr(self, haystack: str, needle: str) -> int:
        return haystack.find(needle)
